src/Orleans.Runtime/Messaging/Gateway.cs

```diff
@@ -59,10 +58,13 @@ public static GrainAddress GetClientActivationAddress(GrainId clientId, SiloAddr
             // Need to pick a unique deterministic ActivationId for this client.
             // We store it in the grain directory and there for every GrainId we use ActivationId as a key
             // so every GW needs to behave as a different "activation" with a different ActivationId (its not enough that they have different SiloAddress)
-            string stringToHash = $"{clientId}{new SpanFormattableIPEndPoint(siloAddress.Endpoint)}{siloAddress.Generation}";
-            Guid hash = Utils.CalculateGuidHash(stringToHash);
-            var activationId = new ActivationId(hash);
-            return GrainAddress.GetAddress(siloAddress, clientId, activationId);
+
+            Span<byte> bytes = stackalloc byte[16];
+            BinaryPrimitives.WriteUInt32LittleEndian(bytes, clientId.Type.GetUniformHashCode());
+            BinaryPrimitives.WriteUInt32LittleEndian(bytes[4..], clientId.Key.GetUniformHashCode());
+            BinaryPrimitives.WriteUInt64LittleEndian(bytes[8..], (uint)siloAddress.GetConsistentHashCode());
```

Comment:
This is leaving 4 bytes empty. This is something which is only called once, upon startup, so I'm not sure optimization is particularly important. Perhaps initializing/invoking SHA256 would slow down startup, but I haven't measured.

Are these optimizations being guided by profiling?

Reply:
It's using `WriteUInt64LittleEndian` to write the last 8 bytes. This is just to avoid allocations and it's probably also faster than anything SHA256 based.

Reply:
I'm sure it's faster than SHA256. I'm not sure it's worth optimizing a one-off call though. There's enough entropy anyway.

Reply:
It's not just optimizing the execution time or allocations, it's also reduced code size to maintain and JIT. I clarfieid the code a bit.

Reply:
I'm ok with it - you're right, it potentially allows SHA256 to be trimmed out / not referenced, for one.